addOns/ascanrules/src/main/java/org/zaproxy/zap/extension/ascanrules/SinkDetectionStorage.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2013 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.ascanrules;
+
+import com.googlecode.concurrenttrees.radix.node.concrete.DefaultCharArrayNodeFactory;
+import com.googlecode.concurrenttrees.radix.node.concrete.voidvalue.VoidValue;
+import com.googlecode.concurrenttrees.radixinverted.ConcurrentInvertedRadixTree;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import org.parosproxy.paros.network.HttpMessage;
+
+public class SinkDetectionStorage {
+
+    private ConcurrentInvertedRadixTree<VoidValue> seenValues;
```

Comment:
can be written over interface.

Reply:
Can you explain a bit more? I don't understand.

Reply:
ConcurrentInvertedRadixTree, i think has one interface also `InvertedRadixTree` so i was saying to write on
```java
InvertedRadixTree<VoidValue> seenValues;
```